phoenix-core/src/it/java/org/apache/phoenix/mapreduce/VerifyReplicationToolIT.java

```diff
@@ -0,0 +1,323 @@
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.mapreduce.Counters;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.phoenix.end2end.BaseUniqueNamesOwnClusterIT;
+import org.apache.phoenix.util.EnvironmentEdgeManager;
+import org.apache.phoenix.util.ReadOnlyProps;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.phoenix.util.TestUtil.TEST_PROPERTIES;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+public class VerifyReplicationToolIT extends BaseUniqueNamesOwnClusterIT {
+    private static final Logger LOG = LoggerFactory.getLogger(VerifyReplicationToolIT.class);
+    private static final String CREATE_USER_TABLE = "CREATE TABLE IF NOT EXISTS %s ( " +
+            " TENANT_ID VARCHAR NOT NULL, USER_ID VARCHAR NOT NULL, AGE INTEGER " +
+            " CONSTRAINT pk PRIMARY KEY ( TENANT_ID, USER_ID ))";
+    private static final String UPSERT_USER = "UPSERT INTO %s VALUES (?, ?, ?)";
+    private static final String UPSERT_SELECT_USERS =
+            "UPSERT INTO %s SELECT TENANT_ID, USER_ID, %d FROM %s WHERE TENANT_ID = ? LIMIT %d";
+    private static final Random RANDOM = new Random();
+
+    private static int tenantNum = 0;
+    private static int userNum = 0;
+    private static String sourceTableName;
+    private static String targetTableName;
+    private List<String> sourceTenants;
+    private String sourceOnlyTenant;
+    private String sourceAndTargetTenant;
+    private String targetOnlyTenant;
+
+    @BeforeClass
+    public static void createTables() throws Exception {
+        NUM_SLAVES_BASE = 2;
+        Map<String,String> props = Maps.newHashMapWithExpectedSize(1);
+        setUpTestDriver(new ReadOnlyProps(props.entrySet().iterator()));
+        Connection conn = DriverManager.getConnection(getUrl());
+        sourceTableName = generateUniqueName();
+        targetTableName = generateUniqueName();
+        // tables will have the same schema, but a different number of regions
+        conn.createStatement().execute(String.format(CREATE_USER_TABLE, sourceTableName));
+        conn.createStatement().execute(String.format(CREATE_USER_TABLE, targetTableName));
+        conn.commit();
+    }
+
+    @Before
+    public void setupTenants() throws Exception {
+        sourceTenants = new ArrayList<>(2);
+        sourceTenants.add("tenant" + tenantNum++);
+        sourceTenants.add("tenant" + tenantNum++);
+        sourceOnlyTenant = sourceTenants.get(0);
+        sourceAndTargetTenant = sourceTenants.get(1);
+        targetOnlyTenant = "tenant" + tenantNum++;
+        upsertData();
+        split(sourceTableName, 4);
+        split(targetTableName, 2);
+        // ensure scans for each table touch multiple region servers
+        ensureRegionsOnDifferentServers(sourceTableName);
```

Comment:
The method assumes that you have only two regions, however you have 4 regions for the first table. If I understand correctly, you are trying to split the table evenly between the 20 rows of 2 tenants and then assigning them on 2 region servers. However the implementation is not working in that manner.

Reply:
See BaseTest.splitSystemCatalog which does something similar, splits the table into multiple regions and moves each of them to their own region server. If you can generalize that method, you can just call it from this test.